### PR TITLE
#4508 writing tests for devToolEpics

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4508)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/store/epics/devTool/fetchTopicDataEpic.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ export default (action$, state$) =>
   action$.pipe(
     ofType(actions.fetchDevToolTopicData.TRIGGER),
     map(action => action.payload),
-    switchMap(values =>
+    exhaustMap(values =>
```
Comment (saivirtue) [inline ohara-manager/client/src/store/epics/devTool/fetchTopicDataEpic.js:29]: To avoid the former request is still on the fly and the later request has been trigger, we use `exhaustMap` to wait for the ongoing observable finished.

File: ohara-manager/client/src/store/epics/devTool/fetchLogEpic.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ import * as selectors from 'store/selectors';
 export default (action$, state$) =>
   action$.pipe(
     ofType(actions.fetchDevToolLog.TRIGGER),
-    switchMap(() =>
+    exhaustMap(() =>
```
Comment (saivirtue) [inline ohara-manager/client/src/store/epics/devTool/fetchLogEpic.js:31]: ditto

File: ohara-manager/client/src/hooks/devToolHooks.js [hunk 3]
```diff
@@ -43,11 +43,16 @@ export const useRefetchDevToolTopicDataAction = () => {
 };
 export const useSetDevToolTopicQueryParams = () => {
   const dispatch = useDispatch();
-  const group = hooks.useTopicGroup();
+  const topicGroup = hooks.useTopicGroup();
```
Comment (konekoya) [inline ohara-manager/client/src/hooks/devToolHooks.js:46]: Any reason for the name changing ❓

File: ohara-manager/client/src/store/epics/devTool/setDevToolLogQueryParamsEpic.js [hunk 4]
```diff
@@ -80,7 +81,7 @@ export default action$ =>
             switchMap(() =>
               scheduled(
                 [
-                  // initial the hostName value
+                  // initial the hostName value when switch to stream
```
Comment (konekoya) [inline ohara-manager/client/src/store/epics/devTool/setDevToolLogQueryParamsEpic.js:84]: Would rephrase this to :)
// provide an initial value when switching to stream

File: ohara-manager/client/src/store/epics/devTool/setDevToolLogQueryParamsEpic.js [hunk 5]
```diff
@@ -96,31 +97,20 @@ export default action$ =>
...
-                      // initial the hostName value
-                      actions.setDevToolLogQueryParams.success({
-                        hostName: '',
-                      }),
-                    ],
-                asapScheduler,
-              ),
-            ),
+            switchMap(() => {
+              const outActions = [
```
Comment (konekoya) [inline ohara-manager/client/src/store/epics/devTool/setDevToolLogQueryParamsEpic.js:101]: Hmm, what does this `outActions` mean...

File: ohara-manager/client/src/hooks/devToolHooks.js (hunk 3 above)
Comment (saivirtue) [reply ohara-manager/client/src/hooks/devToolHooks.js:46]: For consistency with `useSetDevToolLogQueryParams` :)

File: ohara-manager/client/src/store/epics/devTool/setDevToolLogQueryParamsEpic.js (hunk 5 above)
Comment (saivirtue) [reply ohara-manager/client/src/store/epics/devTool/setDevToolLogQueryParamsEpic.js:101]: Epic is "actions in, actions out" 😅 
The `outActions` means the return actions

Comment (konekoya) [reply ohara-manager/client/src/store/epics/devTool/setDevToolLogQueryParamsEpic.js:101]: Ah, okay... need to familiarize myself with these terms
